### PR TITLE
Added sorting features

### LINKED ISSUE #16: Added sorting features
Added sorting based on price
Added sorting based on rating
Added sorting based on price range

### PR DESCRIPTION
**Pull request by Raghav888**
1. Added sorting based on price
2. Added sorting based on rating
3. Added sorting based on price range

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -158,5 +172,6 @@ export const products = [
     productImage: "./assets/comedy_woman.jpg",
     productDescription: `A funny romantic comedy that will make you laugh out loud!‘The voice of modern woman.’ MARIE CLAIRE‘More than funny, it’s true.’ ELLEThe laugh-out-loud Sunday Times bestseller is back - and funnier than ever!Perfect for fans of Outnumbered and Carole Matthews, Fiona writes about life as it really is.After yet another disaster, Lorrie is calling time on online dating. She might be single in her forties, but she’s got a good job, wonderful children and she’s happy`,
```
Comment (Manju-sree) [inline src/backend/db/products.js:173]: good to see you provided ProductDescription clearly .

**Review by Manju-sree — APPROVED** [review]
I like that you used useReducer and useContext in your code, looks fine to merge this PR.

File: src/App.css [hunk 2]
```diff
@@ -12,6 +12,13 @@
     font-size: 1.2rem;
 }
 
+.rating-color {
+    color: #fca500;
```
Comment (rushikesh3299) [inline src/App.css:16]: here you can use var for color if it is repeating again

File: src/components/filterbox/Filterbox.js [hunk 3]
```diff
@@ -48,11 +72,33 @@ export const Filterbox = () => {
           <input
             className="mantra-textbox-classic mantra-highlight-box"
             placeholder=" $499"
+            value={
+              productListstate.minPrice === 0
+                ? undefined
```
Comment (tusharkandpal) [inline src/components/filterbox/Filterbox.js:77]: try avoiding `undefined`

**Review by rushikesh3299 — APPROVED** [review]
Given you a suggestion for css. Everything else is awesome.

File: src/utils/productFilterSort.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+const filterDatabyPrice = ({ minPrice, maxPrice }, productData) => {
+  const returndata = productData.filter(
```
Comment (tusharkandpal) [inline src/utils/productFilterSort.js:2]: instead of storing in a variable you can directly return.

File: src/utils/productFilterSort.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  if (categoryFilter.length >= 1) {
+    const categoryFiltereddata = productData.filter(
+      (product) => categoryFilter.indexOf(product.categoryName) !== -1
+    );
+    return categoryFiltereddata;
+  }
+  return productData;
+};
+const compose = (state, ...args) => {
+  const output = args.reduce((acc, curr) => {
```
Comment (tusharkandpal) [inline src/utils/productFilterSort.js:34]: here too...either just return or give a more relatable name

**Review by tusharkandpal — APPROVED** [review]
well implemented the sorting of products. 👍🏻  

 just one suggestion to look on the default selected radio button on rating filter, if that's expected or not.

**Pull request merged**